llvm_mode/afl-llvm-pass.so.cc

```diff
@@ -146,19 +162,29 @@ bool AFLCoverage::runOnModule(Module &M) {
 
       LoadInst *Counter = IRB.CreateLoad(MapPtrIdx);
       Counter->setMetadata(M.getMDKindID("nosanitize"), MDNode::get(C, None));
-      Value *Incr = IRB.CreateAdd(Counter, ConstantInt::get(Int8Ty, 1));
+      Value *Incr = IRB.CreateAdd(Counter, ConstantInt::get(Int64Ty, 1));
       IRB.CreateStore(Incr, MapPtrIdx)
           ->setMetadata(M.getMDKindID("nosanitize"), MDNode::get(C, None));
 
       /* Set prev_loc to cur_loc >> 1 */
 
       StoreInst *Store =
-          IRB.CreateStore(ConstantInt::get(Int32Ty, cur_loc >> 1), AFLPrevLoc);
+          IRB.CreateStore(ConstantInt::get(Int64Ty, block_counter >> 1), AFLPrevLoc);
       Store->setMetadata(M.getMDKindID("nosanitize"), MDNode::get(C, None));
 
+      // Make a raw_string_ostream to save the content of a block into a string
```

Comment:
This part is what makes the code slow. I tried defining the string_stream once and flushing it every time, but it got slower.

Reply:
maybe you can just stream to file directly?

Reply:
Yes, when I streamed to file it was still slow when I printed the full BB to a raw file, so I thought it's best to go with the cleaner debug file. Writing BB to a stream seems to be slow. When I removed every other change and only kept that, it was still slow.